weaviate/collections/tenants/executor.py

```diff
@@ -467,3 +466,100 @@ def resp(res: Response) -> bool:
+        if self._validate_arguments:
+            _validate_input(
+                _ValidateArgument(
+                    expected=[
+                        str,
+                        Tenant,
+                        Sequence[Union[str, Tenant]],
+                    ],
+                    name="tenant",
+                    value=tenant,
+                )
+            )
+        if isinstance(tenant, str) or isinstance(tenant, Tenant):
+            tenants = [
+                TenantUpdate(
+                    name=tenant.name if isinstance(tenant, Tenant) else tenant,
+                    activity_status=activity_status,
+                )
+            ]
+        else:
+            tenants = [
+                TenantUpdate(
+                    name=t.name if isinstance(t, Tenant) else t,
+                    activity_status=activity_status,
+                )
+                for t in tenant
+            ]
+        return self.__update(tenants=tenants)
+
+    def activate(
+        self, tenant: Union[TenantInputType, Sequence[TenantInputType]]
+    ) -> executor.Result[None]:
+        """Activate the specified tenants for this collection in Weaviate.
+
+        The collection must have been created with multi-tenancy enabled.
+
+        Args:
+            tenant: A tenant name, `wvc.config.tenants.Tenant` object, or a list of tenants names
+                and/or `wvc.config.tenants.Tenant` objects to activate for the given collection.
+
+        Raises:
+            weaviate.exceptions.WeaviateConnectionError: If the network connection to Weaviate fails.
+            weaviate.exceptions.UnexpectedStatusCodeError: If Weaviate reports a non-OK status.
+            weaviate.exceptions.WeaviateInvalidInputError: If `tenant` is not a list of `wvc.Tenant` objects.
+        """
+        self.__update_tenant_activity_status(
+            tenant=tenant,
+            activity_status=TenantUpdateActivityStatus.ACTIVE,
+        )
+
+    def deactivate(
+        self, tenant: Union[TenantInputType, Sequence[TenantInputType]]
+    ) -> executor.Result[None]:
+        """Deactivate the specified tenants for this collection in Weaviate.
+
+        The collection must have been created with multi-tenancy enabled.
+
+        Args:
+            tenant: A tenant name, `wvc.config.tenants.Tenant` object, or a list of tenants names
+                and/or `wvc.config.tenants.Tenant` objects to deactivate for the given collection.
+
+        Raises:
+            weaviate.exceptions.WeaviateConnectionError: If the network connection to Weaviate fails.
+            weaviate.exceptions.UnexpectedStatusCodeError: If Weaviate reports a non-OK status.
+            weaviate.exceptions.WeaviateInvalidInputError: If `tenant` is not a list of `wvc.Tenant` objects.
+        """
+        self.__update_tenant_activity_status(
+            tenant=tenant,
+            activity_status=TenantUpdateActivityStatus.INACTIVE,
+        )
+
+    def offload(
```

Comment:
Could it be confusing for users if they pass in a tenant with status Active and then it gets offloaded? 🤔  I think it is ok but wanted to bring that up

Reply:
Hmm, good point 🤔 By allowing `Tenant` here I was intending for this pattern to work easily:
```python
tenant = collection.tenants.create("t1")
collection.tenants.deactivate(tenant)
```

Reply:
we could update the `tenant` status to reflect the new state?